test/functional/test_runner.py

```diff
@@ -56,12 +56,13 @@
         stderr_mode = ctypes.c_int32()
         kernel32.GetConsoleMode(stderr, ctypes.byref(stderr_mode))
         kernel32.SetConsoleMode(stderr, stderr_mode.value | ENABLE_VIRTUAL_TERMINAL_PROCESSING)
-    # primitive formatting on supported
-    # terminal via ANSI escape sequences:
-    BOLD = ('\033[0m', '\033[1m')
-    GREEN = ('\033[0m', '\033[0;32m')
-    RED = ('\033[0m', '\033[0;31m')
-    GREY = ('\033[0m', '\033[1;30m')
+    if sys.stdout.isatty():
```

Comment:
please call the `sys.stdout.isatty()` only once and assign it to a variable (`use_term_control` or something like that) if you use it later
I'd like to add an argument to override this, there are cases where one might want to redirect the output to something that doesn't identify as a tty but still supports terminal control sequences
(e.g. when calling the tests over SSH)

Reply:
Yeah my suggestion was to use isatty value as the option default value.

Reply:
Good points. Now addressed. Please re-review :-)